### PR TITLE
Update parity-codec version

### PR DESCRIPTION
**Pull request by DarkEld3r**
In progress because it references https://github.com/paritytech/parity-common/pull/178 branch. Will be updated when all issues are sorted.

### CODE REVIEW

File: core/finality-grandpa/src/environment.rs [hunk 1]
```diff
@@ -637,7 +637,7 @@ where
 		round: u64,
 		state: RoundState<Block::Hash, NumberFor<Block>>,
 		base: (Block::Hash, NumberFor<Block>),
-		votes: Vec<SignedMessage<Block>>,
+		votes: &HistoricalVotes<Block::Hash, NumberFor<Block>, Self::Signature, Self::Id>,
```
Comment (bkchr) [inline core/finality-grandpa/src/environment.rs]: Why do we need this changes?

Comment (DarkEld3r) [reply core/finality-grandpa/src/environment.rs]: It is an implementation of the `Environment` trait from `finality-grandpa`. The trait definition is changed, so the implementation needs to be updated.

Comment (bkchr) [reply core/finality-grandpa/src/environment.rs]: Ahh you updated grandpa, makes sense!

**Comment by DarkEld3r** [comment]
`4.1` version has been released (https://github.com/paritytech/parity-scale-codec/pull/104), so the pull request has been updated too.

**Pull request merged**